### ISSUE TITLE
Getting Error in Analyzer code

### ISSUE THREAD

**Issue by padmadighe** [rich-newman/typescript-analyzer#7]
Hi,

We are getting error in typescript analyzer code.

6/13/2017 4:01:25 PM: System.Exception: TsLint call on build timed out.  Timeout is 5 seconds.
   at WebLinter.NodeServer.<CallServer>d__8.MoveNext() in C:\Source\typescript-analyzer\src\WebLinter\NodeServer.cs:line 37
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at WebLinter.LinterBase.<RunProcess>d__28.MoveNext() in C:\Source\typescript-analyzer\src\WebLinter\Linting\Linters\LinterBase.cs:line 77
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at WebLinter.LinterBase.<Lint>d__27.MoveNext() in C:\Source\typescript-analyzer\src\WebLinter\Linting\Linters\LinterBase.cs:line 58
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at WebLinter.LinterBase.<Run>d__26.MoveNext() in C:\Source\typescript-analyzer\src\WebLinter\Linting\Linters\LinterBase.cs:line 53
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at WebLinter.LinterFactory.<LintAsync>d__5.MoveNext() in C:\Source\typescript-analyzer\src\WebLinter\Linting\LinterFactory.cs:line 54
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at WebLinterVsix.LinterService.<LintAsync>d__4.MoveNext() in C:\Source\typescript-analyzer\src\WebLinterVsix\LinterService.cs:line 67

There is no change in configuration but somehow stopped working.
Thanks
Padma

**Comment by rich-newman** [comment]
You're going to have to give me a little more to work with: does it happen every time?  does your linting actually take more than 5 seconds to run?  do you have any other pre- or post-build events?  is there anything else that could be blocking the UI thread during a build?

**Comment by padmadighe** [comment]
Yes it happens every time with build and even with run typescript analyzer. We have ng build command that runs on 'after build' event. This event is there 
 even when it worked before.  Today I removed and still was getting same error.

Also, major problem is build does not fail when analyzer fails.

**Comment by rich-newman** [comment]
OK well nothing has changed: there's been no release or anything.  The analyzer spins up a local node webserver to call TSLint, something I'm thinking of changing.  The webserver hangs around whilst Visual Studio is open, and if VS crashes it can hang around longer than that.  You'll see it as a node process in Task Manager.  On a build it has to block while that call is made, and the timeout is there to ensure it doesn't bring down Visual Studio if the call never returns or deadlocks for some reason.  I'm assuming you've tried restarting VS, and then restarting the machine: if not, try those, and try running it in a new console app to see if the same thing happens.

**Comment by padmadighe** [comment]
I restarted both VS and machine but didn't work.  This workaround won't be acceptable as skipping style check is a big red flag for us.
Is it possible to fail build if analyzer fails?

 I'm attaching sample project where I can replicate the issue. Please change extension from txt to rar

[SampleApp.txt](https://github.com/rich-newman/typescript-analyzer/files/1073551/SampleApp.txt)

**Comment by rich-newman** [comment]
I tried your sample app and it works fine for me, see below.  That makes things difficult.
![image](https://user-images.githubusercontent.com/17277125/27228039-efdbd792-529d-11e7-8e3f-536fd70f6f3c.png)

**Comment by padmadighe** [comment]
Sample app is not working on multiple machines. Also, sometimes, analyzer will not even run on build.
I think below error is the root cause, in what scenarios this error will popup?
Unable to lint: webserver not correctly initialized

**Comment by rich-newman** [comment]
That happens if the webserver process has died for some reason when you try to lint synchronously, which is what you do on a build.  The webserver process is initialized at startup and shouldn't die.

I have been working on a version of the analyzer that doesn't use a webserver.  Instead it uses Tomasz Janczuk's [edge project](https://github.com/tjanczuk/edge) to call JavaScript from C#.  I've created a build of that, which is on the link below.  

The code is in the this Github repo, on the edge branch.  It's quite a bit faster to lint than the existing version.  It's had some testing, but I haven't released it to VS Marketplace because the existing one seems to work fine.  Apart from for you.

Can you try this and see if it fixes your problem?

[TypeScript Analyzer Edge v1.41](https://www.dropbox.com/s/3n4qpc0aqirnwmz/TypeScriptAnalyzerEdge.vsix?dl=0)

Just click to download it and then doubleclick the .vsix file to install it.

**Comment by padmadighe** [comment]
Hi, 

Not able to install it. Looks like issue due to different install location. Attaching log file
[VSIXInstaller_0049777a-d433-4bf1-86e7-4a678261371f.txt](https://github.com/rich-newman/typescript-analyzer/files/1097035/VSIXInstaller_0049777a-d433-4bf1-86e7-4a678261371f.txt)

**Comment by rich-newman** [comment]
I don't know why that test version won't install for you.  As you can see the log's saying it's looking for devenv.exe on the C: drive.  It clearly already knows your installation is on the G: drive as it says so earlier in the log.  I tested by installing Enterprise on a different drive to make sure the .vsix file would install there and it worked fine for me.

I don't think it's worth spending a lot of time on that though, it's only a test file.

Otherwise I have good news and bad news.  The good news is I looked again at the build logic, found some problems, and fixed them.  In particular I realized that if you didn't have the Error List open and did a build immediately after opening a solution it would fail to run the linter correctly.  I've fixed that, and I've also put some more logic around ensuring the node process is running correctly when you do a build.

There's a new release with those changes in, version 1.5.

The bad news is that none of the problems I found gave the error message you are getting, so I'm unsure if I've actually fixed your problem.  Which I'm still unable to replicate.

Can you upgrade and test with the new version when you get a moment please?

**Comment by padmadighe** [comment]
1. I upgraded analyzer didn't work .
2. I uninstalled and then reinstalled but still didn't work.
3. I then uninstalled all the related extensions but still didn't work.
4. Finally I reinstalled VS2017 and now  Analyzer is working :-)

I think some other extension or VS 2017 feature is blocking analyzer.  We will wait for issue's recurrence for next 15 days . If doesn't happen  will close the issue.
Thanks for your help :-)
